Automattic/wp-calypso
# State: Introduce site monitor actions, reducer, selectors & query components

This PR introduces actions, reducer, selectors and query components for managing site monitor settings. It consists of:

* `requestSiteMonitorSettings` action - used to fetch the monitor settings of a site
* `updateSiteMonitorSettings` action - used to update the monitor settings of a site
* `monitor` reducer - used to handle site monitor settings and requests
* `getSiteMonitorSettings` selector - used to retrieve the monitor settings from the state
* `isRequestingSiteMonitorSettings` selector - whether currently retrieving the monitor settings for a certain site
* `isUpdatingSiteMonitorSettings` selector - whether currently updating the monitor settings for a certain site
* `QuerySiteMonitorSettings` query component - used to handle network requests for fetching the monitor settings of a site.
* Tests for all of the above.

These will be used to replace the `JetpackSite.prototype.fetchMonitorSettings` and `JetpackSite.prototype.updateMonitorSettings` method and the site attributes that they manage (`monitorSettings` and `fetchingMonitorSettings`) as we migrate away from `sites-list`. 

To test:
* Checkout this branch
* Verify all tests pass:
  * `npm run test-client client/state/sites/monitor/`
  * `npm run test-client client/state/selectors/test/get-site-monitor-settings.js`
  * `npm run test-client client/state/selectors/test/is-requesting-site-monitor-settings.js`
  * `npm run test-client client/state/selectors/test/is-updating-site-monitor-settings.js`
* Insert the `<QuerySiteMonitorSettings siteId={ siteId } />` component somewhere and verify it performs the expected network requests.
